### PR TITLE
[bug] fixed valid-jsdoc rule (closes #227)

### LINKED ISSUE #227: TypeScript abstract method return JSDoc
### Bug Report

- __TSLint version__: 5.2.0
- __TypeScript version__: 2.2.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
abstract class Foo { 
  /**
   * @abstract
   * @returns {string} some string.
   */
  public abstract foo(): string;
}
```

with `tslint.json` configuration:

```js
module.exports = {
  rules: {
    ...
    'valid-jsdoc': [true, {
      prefer: {
        return: 'returns'
      },
      preferType: {
        object: 'Object',
        array: 'Array',
        function: 'Function'
      },
      requireReturn: false,
      requireParamDescription: true,
      requireReturnDescription: true
    }],
    ...
  }
};
```

#### Actual behavior

```
ERROR: Foo.ts[1, 21]: unexpected @returns tag; function has no return statement
```

#### Expected behavior

Clean output

### PR DESCRIPTION
**Pull request by RecuencoJones**

### CODE REVIEW

File: src/rules/validJsdocRule.ts [hunk 1]
```diff
@@ -336,7 +337,11 @@ class ValidJsdocWalker extends Lint.RuleWalker {
         case 'returns':
           hasReturns = true;
 
-          if (!OPTIONS.requireReturn && !fn.returnPresent && tag.type.name !== 'void' && tag.type.name !== 'undefined') {
+          if (fn.node.modifiers) {
```
Comment (ajafff) [inline src/rules/validJsdocRule.ts]: `isAbstract = Lint.hasModifier(fn.node.modifiers, ts.SyntaxKind.AbstractKeyword`
No need to check for undefined

Comment (RecuencoJones) [inline src/rules/validJsdocRule.ts]: Wasn't aware of that helper method, fixing...

**Comment by RecuencoJones** [comment]
@ajafff, @jmlopez-rod do you have an estimation regarding when this PR could be merged?

**Comment by jmlopez-rod** [comment]
@RecuencoJones released in 4.1.1, thanks for the fix.

**Pull request merged**